ledger/complete/compactor.go

```diff
@@ -45,16 +45,17 @@ type checkpointResult struct {
 // This will be resolved automaticaly after the forest LRU Cache
 // (code outside checkpointing) is replaced by something like a FIFO queue.
 type Compactor struct {
-	checkpointer       *realWAL.Checkpointer
-	wal                realWAL.LedgerWAL
-	trieQueue          *realWAL.TrieQueue
-	logger             zerolog.Logger
-	lm                 *lifecycle.LifecycleManager
-	observers          map[observable.Observer]struct{}
-	checkpointDistance uint
-	checkpointsToKeep  uint
-	stopCh             chan chan struct{}
-	trieUpdateCh       <-chan *WALTrieUpdate
+	checkpointer                           *realWAL.Checkpointer
+	wal                                    realWAL.LedgerWAL
+	trieQueue                              *realWAL.TrieQueue
+	logger                                 zerolog.Logger
+	lm                                     *lifecycle.LifecycleManager
+	observers                              map[observable.Observer]struct{}
+	checkpointDistance                     uint
+	checkpointsToKeep                      uint
+	stopCh                                 chan chan struct{}
+	trieUpdateCh                           <-chan *WALTrieUpdate
+	triggerCheckpointOnNextSegmentFinishCh <-chan interface{} // to trigger checkpoint manually
```

Comment:
Would be better to use boolean - so it can be easily turned off and status checked, I don't think there is any benefit to use channel.

Reply:
Good suggestion, close in favor of this https://github.com/onflow/flow-go/pull/3010